treex/nn/recurrent.py

```diff
@@ -0,0 +1,183 @@
+import typing as tp
+from functools import partial
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+import treeo as to
+from flax.linen import recurrent as flax_module
+
+from treex import types
+from treex.key_seq import KeySeq
+from treex.module import Module, next_key
+from treex.nn.linear import Linear
+
+CallableModule = tp.Callable[..., jnp.ndarray]
+
+
+class GRU(Module):
+    """Gated Recurrent Unit - Cho et al. 2014
+
+    `GRU` is implemented as a wrapper on top of `flax.linen.GRUCell`, providing higher level
+    functionality and features similar to what can be found in that of `tf.keras.layers.GRU`.
+    """
+
+    gate_fn: CallableModule = flax_module.sigmoid
+    activation_fn: CallableModule = flax_module.tanh
+    kernel_init: tp.Callable[
+        [flax_module.PRNGKey, flax_module.Shape, flax_module.Dtype], flax_module.Array
+    ]
+    recurrent_kernel_init: tp.Callable[
+        [flax_module.PRNGKey, flax_module.Shape, flax_module.Dtype], flax_module.Array
+    ]
+    bias_init: tp.Callable[
+        [flax_module.PRNGKey, flax_module.Shape, flax_module.Dtype], flax_module.Array
+    ]
+    initial_state_init: tp.Callable[
+        [flax_module.PRNGKey, flax_module.Shape, flax_module.Dtype], flax_module.Array
+    ]
+    params: tp.Dict[str, tp.Dict[str, flax_module.Array]] = types.Parameter.node()
+    last_state: flax_module.Array = types.BatchStat.node()
```

Comment:
Unsure if this should be a `BatchStat` node, these are intended to be synced during distributed training. I think `types.Cache.node()` is a better fit.